### PR TITLE
chore: upgraded from JSDOC to TSDOC config

### LINKED ISSUE #699: Switch from jsdoc to typedoc

### PR DESCRIPTION
**Pull request by slashexx**
## Description

This pull request changes the following:

* Updated the documentation generation process to leverage TypeDoc instead of JSDoc.
* Configured ESLint to include `eslint-plugin-tsdoc` for improved TSDoc syntax checking.

### Related Issues

* Closes #699

### CODE REVIEW

File: docs/Taskfile.yaml [hunk 1]
```diff
@@ -63,9 +63,9 @@ tasks:
       - hugo config
       - hugo --gc --config hugo.toml --minify --baseURL "$HUGO_BASE_URL/"
 
-  build:jsdoc:
+  build:typedoc:
     cmds:
-      - jsdoc -c jsdoc.conf.json
+      - typedoc --out ./content/Classes ../src
```
Comment (jeromy-cannon) [inline docs/Taskfile.yaml:68]:
```suggestion
      - cd .. && typedoc --out docs/public/static/Classes --entryPoints ./solo.ts --entryPoints ./src/index.ts --entryPointStrategy expand ./src
```

File: docs/Taskfile.yaml [hunk 2]
```diff
@@ -77,17 +77,17 @@ tasks:
   install:
     cmds:
       - task: "install:hugo"
-      - task: "install:jsdoc"
+      - task: "install:typedoc"
```
Comment (jeromy-cannon) [inline docs/Taskfile.yaml:80]:
```suggestion
      - task: "install:typedoc"
      - task: "install:solo"
```

**Comment by slashexx** [comment]
Apologies, I got a little late.

Seems it's still failing the hugo build ?

File: docs/Taskfile.yaml [hunk 3]
```diff
@@ -77,17 +77,17 @@ tasks:
...
-  install:jsdoc:
+  install:typedoc:
     status:
-      - command -v jsdoc
+      - command -v typedoc
     cmds:
-      - npm i -g jsdoc
-      - jsdoc --version
+      - npm i -g typedoc
+      - typedoc --version
```
Comment (jeromy-cannon) [inline docs/Taskfile.yaml:94]:
```suggestion
      - typedoc --version

  install:solo:
    cmds:
      - cd ..
      - npm install
```

**Review by jeromy-cannon — APPROVED** [review]

**Comment by jeromy-cannon** [comment]
> Apologies, I got a little late.
> 
> Seems it's still failing the hugo build ?

No problem, I wasn't sure when you would next be available.  The Hugo build is passing now.  The update readme fails, because of access.  I'll override and merge it soon. 

Thank you!

**Review by jeromy-cannon — APPROVED** [review]

**Pull request merged**